spec/decorator_spec.rb

```diff
@@ -0,0 +1,33 @@
+require_relative '../basedecorator'
+require_relative '../capitalizedecorator'
+require_relative '../trimmerdecorator'
+
+describe BaseDecorator do
+  before :each do
+    @base_decorator = BaseDecorator.new('Ichsan')
+  end
+
+  it 'should return an instance of BaseDecorator' do
+    expect(@base_decorator).to be_an_instance_of BaseDecorator
+  end
+end
+
+describe CapitalizeDecorator do
+  before :each do
+    @capitalize_decorator = CapitalizeDecorator.new('Ichsan')
+  end
+
+  it 'should return an instance of CapitalizeDecorator' do
+    expect(@capitalize_decorator).to be_an_instance_of CapitalizeDecorator
+  end
+end
+
+describe TrimmerDecorator do
+  before :each do
+    @trimmer_decorator = TrimmerDecorator.new('Ichsan')
+  end
+
+  it 'should return an instance of TrimmerDecorator' do
+    expect(@trimmer_decorator).to be_an_instance_of TrimmerDecorator
+  end
+end
```

Comment:
- [ ] Based on the requirements, each class should have a file hence we'll required to kindly refactor these codes into different files since they are difference classes.